stl/inc/ranges

```diff
@@ -728,6 +735,489 @@ namespace ranges {
         using all_t = decltype(all(_STD declval<_Rng>()));
     } // namespace views
 
+    // CLASS TEMPLATE ranges::transform_view
+    // clang-format off
+    template <class _Rng, class _Fn> // TRANSITION, LLVM-47414
+    concept _Can_const_transform = range<const _Rng> && regular_invocable<const _Fn&, range_reference_t<const _Rng>>;
+
+#if _ITERATOR_DEBUG_LEVEL == 0
+#define _NOEXCEPT_IDL0(...) noexcept(__VA_ARGS__)
+#else
+#define _NOEXCEPT_IDL0(...)
+#endif // _ITERATOR_DEBUG_LEVEL == 0

```

Comment:
I don't like the idea of `noexcept` being dependent on debug level. If debug machinery throws the program should just terminate. It also makes the code harder to understand.

Reply:
I'm also surprised to see this at the last minute. In `vector` and `string` we've allowed debug allocations to slam into `noexcept` (which causes headaches occasionally, but there are no good choices there).

Reply:
I could live with "just `terminate`" - it certainly was simpler - but I expect we'll hear complaints about it. I reserve the right to say "I told you so!" when that day comes ;)

Reply:
After thinking about it a little more, termination does seem problematic - we should probably err on the side of not strengthening `noexcept`, or making the precondition checks controlled by the `noexcept`ness of the operations they want to invoke. @CaseyCarter will file a tracking issue to investigate this (due to time constraints we had to merge this for 16.8 Preview 3 as-is). Thanks for finding this, @AdamBucior!

Reply:
Stephan and I went around in circles a bit about how to resolve this and decided to merge as-is (since this is already merged internally) and file an issue to cleanup this conditional `noexcept` in a followup change. I'll file that issue after this merges - I'd like to be able to link to the significant bits of code - and post the issue number here to close the loop.